DOC-1974: Added TINY-9746 to the release notes. (#2748)
tinymce/tinymce-docs: modules/ROOT/pages/6.5-release-notes.adoc

{productname} 6.5 also includes the following bug fixes:

> === On iOS Safari, Korean characters will no longer merge onto the previous line after inserting a newline by pressing Enter.

Reviewer: If we're trying to match the title exactly to the changelog entry, this is the latest version.
```suggestion
=== On iOS Safari, Korean characters were merging onto the previous line upon typing after inserting a newline by pressing Enter.
```

> === On iOS Safari, Hangul (Korean) characters will no longer merge onto the previous line after inserting a newline by pressing Enter.

Reviewer: As I found out in my follow-up PR for this ticket, Hangul is only a category of Korean characters. I've modified that changelog entry already but this must have been transferred over before that happened.
```suggestion
=== On iOS Safari, Korean characters will no longer merge onto the previous line after inserting a newline by pressing Enter.
```

Author: Sounds good, ill update the incoming changes to the changelog for 6.5

//#TINY-9746

In previous versions of {productname}, in iOS Safari, a bug has been fixed regarding the handling of Korean characters.

> Previously, when a newline was inserted by pressing the Enter key, Korean characters would merge onto the previous line instead of starting a new line. This occurred because iOS Safari composed Korean characters by deleting the previous partial character and inserting the composed character, continuing until the Enter key was pressed. Even though {productname} has its own Enter key handler, the native Enter key event was suppressed, causing the bug.

Reviewer: ```suggestion
Previously, when a newline was inserted by pressing the Enter key and the user continued typing in Korean without moving the caret, Korean characters typed on the newline would merge onto the previous line and the newline would be removed. This occurred because iOS Safari composed Korean characters by deleting the previous partial character and inserting the composed character, continuing until a native Enter `keypress` event had been fired. However, {productname} has its own Enter key handler on `keydown`, so the native Enter `keypress` event was suppressed, preventing iOS Safari from stopping Korean character composition as needed and causing the bug.
```

> {productname} 6.5, now saves a bookmark while adding an undo level when the caret is positioned after a Korean character and the Enter key is pressed. This allows the native Enter key event to fire and end the composition session. Then, on the keyup event, the effects of the Enter key (creating a newline) are undone, the original selection is restored, and {productname}'s Enter key handler is called. As a result, iOS Safari no longer attempts to compose Korean characters across newlines, resolving the issue.

Reviewer: ```suggestion
In {productname} 6.5,  a bookmark is saved and an undo level is added on the Enter `keydown` event when the user is on iOS Safari and the caret is detected to be positioned after a Korean character. This allows the native Enter `keypress` event to fire and end the composition session. Then, on the Enter `keyup` event, the effects of the native Enter keypress (ex: creating a newline) are undone, the original selection is restored via the bookmark, and {productname}'s own Enter key handler is called. As a result, iOS Safari no longer attempts to compose Korean characters across newlines, resolving the issue.
```

> === On iOS Safari, Korean characters were merging onto the previous line upon typing after inserting a newline by pressing Enter.
> //#TINY-9746
> 
> In previous versions of {productname}, in iOS Safari, a bug has been fixed regarding the handling of Korean characters.
> 
> Previously, when a newline was inserted by pressing the Enter key and the user continued typing in Korean without moving the caret, Korean characters typed on the newline would merge onto the previous line and the newline would be removed. This occurred because iOS Safari composed Korean characters by deleting the previous partial character and inserting the composed character, continuing until a native Enter `keypress` event had been fired. However, {productname} has its own Enter key handler on `keydown`, so the native Enter `keypress` event was suppressed, preventing iOS Safari from stopping Korean character composition as needed and causing the bug.
> 
> In {productname} 6.5,  a bookmark is saved and an undo level is added on the Enter `keydown` event when the user is on iOS Safari and the caret is detected to be positioned after a Korean character. This allows the native Enter `keypress` event to fire and end the composition session. Then, on the Enter `keyup` event, the effects of the native Enter keypress (ex: creating a newline) are undone, the original selection is restored via the bookmark, and {productname}'s own Enter key handler is called. As a result, iOS Safari no longer attempts to compose Korean characters across newlines, resolving the issue.

Reviewer 2: ```suggestion
=== In Safari running on iOS, Hangul characters merged into the previous line upon typing after pressing Enter.
//#TINY-9746

Previously, in Safari running on iOS, when a newline was inserted (by pressing the Enter key) and the user continued typing without moving the insertion point, the characters typed on the newline merged onto the previous line and the newline was removed.

This occurred because Safari on iOS composes Hangul syllabic blocks by deleting the individual jaso glyphs and inserting the expected block glyph, continuing until a native Enter `keypress` event is fired.

However, {productname} has its own Enter key handler on `keydown`. In this circumstance, therefore, the native `keypress` event is suppressed. This causes Safari on iOS to continue composing the syllabic block across the newline by firing a `deleteContentBackward` input event to delete the newline and then inserting the block on the previous line.

For {productname} 6.5, when the user is in Safari on iOS and the insertion point is detected as positioned after a Korean character, on Enter key keydown;

. a bookmark is saved and an undo level is added; and
. `preventDefault` is **not** executed to still allow the browser-native Enter key `keypress` event to fire.

This prompts Safari on iOS to end the syllabic block creation session for the previous character.

Then, on the Enter key `keyup` event;

. the effects of the native Enter key `keypress` (usually the creation of a newline) is undone;
. the original selection is restored via the saved bookmark; and
. the {productname}-native Enter key handler is called.

This ensures that, when running in Safari on iOS, Safari no longer attempts to compose syllabic blocks across newlines, preventing the unwanted behavior from occurring.
```

Re-write and re-structure after consultation re nomenclature.

{productname} 6.5 also includes the following bug fixes:

=== In Safari running on iOS, Hangul characters merged into the previous line upon typing after pressing Enter.
//#TINY-9746

Previously, in Safari running on iOS, when a newline was inserted (by pressing the Enter key) and the user continued typing without moving the insertion point, the characters typed on the newline merged onto the previous line and the newline was removed.

> This occurred because Safari on iOS composes Hangul syllabic blocks by deleting the individual jaso glyphs and inserting the expected block glyph, continuing until a native Enter `keypress` event is fired.

Reviewer 2: ```suggestion
This occurred because Safari on iOS composes Hangul syllabic blocks by deleting the individual jaso glyphs and inserting the expected block glyph, continuing until a browser-native Enter `keypress` event is fired.
```

> However, {productname} has its own Enter key handler on `keydown`. In this circumstance, therefore, the native `keypress` event is suppressed. This causes Safari on iOS to continue composing the syllabic block across the newline by firing a `deleteContentBackward` input event to delete the newline and then inserting the block on the previous line.

Reviewer 2: ```suggestion
However, {productname} has its own Enter key handler on `keydown`. In this circumstance, therefore, the native `keypress` event was suppressed. This caused Safari on iOS to continue composing the syllabic block across the newline by firing a `deleteContentBackward` input event to delete the newline and then inserting the block on the previous line.
```

> For {productname} 6.5, when the user is in Safari on iOS and the insertion point is detected as positioned after a Korean character, on Enter key keydown;

Reviewer 2: ```suggestion
For {productname} 6.5, when the user is in Safari on iOS and the insertion point is detected as positioned after a Korean character, on Enter key `keydown`;
```

> . a bookmark is saved and an undo level is added; and

Reviewer 2: ```suggestion
. a bookmark is saved;
. an undo level is added; and
```
